python/ray/tests/test_tempfile.py

```diff
@@ -37,6 +37,8 @@ def test_conn_cluster():
         "When connecting to an existing cluster, "
         "temp_dir must not be provided.")
 
+    ray.shutdown()
```

Comment:
This is pretty strange since the `ray.init` is failing above. Any idea why this would be non-deterministic or why this error would happen? Also, probably preferable to pass `shutdown_only` into the test.

Reply:
Yes, this was more of a test.

This does fix the issue. It appears that somehow ray is still being initialized even though the exception is getting thrown, but it doesn't happen all the time (See master build history, sometimes rerunning can fix the issue). I do not know why